tools/scripts/generic.mk

```diff
@@ -127,9 +127,13 @@ endif
 #------------------------------------------------------------------------------
 
 ifeq 'xilinx' '$(PLATFORM)'
-HARDWARE := $(filter %.hdf, $(HARDWARE))
+TMP_HW := $(filter %.xsa, $(HARDWARE))
+ifeq '' '$(TMP_HW)'
+TMP_HW := $(filter %.hdf, $(HARDWARE))
+endif
+HARDWARE := $(TMP_HW)
```

Comment:
Since the makefile is static, after the first comparison (with xsa) the second else branch (that will search for a hdf) won't be accessible. This is the reason the same variable can't be assigned, compared, assigned, and compared again.

Reply:
IF you use the same variable twice, the second branch will never be true